src/shared/components/svg-icon.js

```diff
@@ -6,7 +6,12 @@ import propTypes from 'prop-types';
 /**
  * Object mapping icon names to SVG markup.
  *
- * @typedef {{[name: string]: string}} IconMap
+ * @typedef {Object.<string,string>} IconMap
```

Comment:
The old/new `IconMap` types here have the same meaning, but the `Object.<KeyType,ValueType>` syntax is what I've been using elsewhere.